### PR TITLE
Create new slot for calendar day content

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #128, #147, #159

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Create new slot that allows users to put content into each day cell in the calendar
* Style the content within the day slot
* Create examples to demonstrate all of our use cases for this slot (text and `auro-icon`)
* Update repo to support node 20
* Update design token prefixes from `--auro` to `--ds`

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Unable to validate UI due to API page not formatting correctly.

File: demo/api.md [hunk 1]
```diff
@@ -738,6 +738,187 @@ Only for use with the `range` attribute. Sets the label for the second input.
 <!-- AURO-GENERATED-CONTENT:END -->
 </auro-accordion>
 
+### Day Slot
+
+Each day within the calendar has a slot that allows a user to add custom content. The user will add their slot content withtin the `auro-datepicker`, with the slot name needing to be in the format of `date_{month}_{day}_{fullYear}`. **Note**: When the month is a single digit, a zero will need to be added to the front of the month. For example, January would be `01` and not `1`.
+
+In order to style your slot content, we recommend using in-line styles on your component, as the day slot content is passed down multiple levels within the component. Another way to style the component is to target the `part="daySlot"` attribute that is added to the container around the slot content.
```
Comment (blackfalcon) [inline demo/api.md:745]:
```suggestion
For `<slot>` styling, it is recommended to use in-line styles. 

... WE NEED A HTML EXAMPLE HERE ....

Another option is to target the `daySlot` CSS Part. This will target the HTML node surrounding the slotted content. 

... WE NEED A CSS EXAMPLE HERE ....
```

File: demo/api.md [hunk 2]
```diff
@@ -738,6 +738,187 @@ Only for use with the `range` attribute. Sets the label for the second input.
...
+
+**Disclaimer**: At this time, we are only supporting the use of text and `auro-icon` as the content being used in the slot.
```
Comment (blackfalcon) [inline demo/api.md:747]:
```suggestion
**Disclaimer**: This element only supports the use of default text or Auro icons to be slotted in the calendar at this time. 
```

File: demo/api.md [hunk 3]
```diff
@@ -738,6 +738,187 @@ Only for use with the `range` attribute. Sets the label for the second input.
...
 
+### Day Slot
+
+Each day within the calendar has a slot that allows a user to add custom content. The user will add their slot content withtin the `auro-datepicker`, with the slot name needing to be in the format of `date_{month}_{day}_{fullYear}`. **Note**: When the month is a single digit, a zero will need to be added to the front of the month. For example, January would be `01` and not `1`.
+
+In order to style your slot content, we recommend using in-line styles on your component, as the day slot content is passed down multiple levels within the component. Another way to style the component is to target the `part="daySlot"` attribute that is added to the container around the slot content.
+
+**Disclaimer**: At this time, we are only supporting the use of text and `auro-icon` as the content being used in the slot.
+
+In this example, we use text content within the day slot to display prices for each day.
```
Comment (blackfalcon) [inline demo/api.md:749]:
```suggestion
In the following example please see how pricing content is threaded into the elements `<slot>`. 
```

File: demo/api.md [hunk 4]
```diff
@@ -738,6 +738,187 @@ Only for use with the `range` attribute. Sets the label for the second input.
...
+  <span slot="date_01_01_2024">$202</span>
+  <span slot="date_01_05_2024">$3022</span>
+</auro-datepicker>
+```
+<!-- AURO-GENERATED-CONTENT:END -->
+</auro-accordion>
+
+#### Day Slot Range w/ Text
+
+Day slot content used with `range` support.
```
Comment (blackfalcon) [inline demo/api.md:820]:
```suggestion
The following example illustrates slotted content in combination with the `range` support. 
```

File: demo/api.md [hunk 5]
```diff
@@ -738,6 +738,187 @@ Only for use with the `range` attribute. Sets the label for the second input.
...
+  <span slot="date_01_01_2024">$202</span>
+  <span slot="date_01_05_2024">$3022</span>
+</auro-datepicker>
+```
+<!-- AURO-GENERATED-CONTENT:END -->
+</auro-accordion>
+
+#### Day Slot w/ Icon
+
+In this example, we show `auro-icon` being used as day slot content.
```
Comment (blackfalcon) [inline demo/api.md:891]:
```suggestion
The following example illustrates how the `<auro-icon>` element can be used as slotted content. 
```

File: docs/partials/api.md [hunk 6]
```diff
@@ -351,6 +351,61 @@ Only for use with the `range` attribute. Sets the label for the second input.
...
+### Day Slot
+
+Each day within the calendar has a slot that allows a user to add custom content. The user will add their slot content withtin the `auro-datepicker`, with the slot name needing to be in the format of `date_{month}_{day}_{fullYear}`. **Note**: When the month is a single digit, a zero will need to be added to the front of the month. For example, January would be `01` and not `1`.
+
+In order to style your slot content, we recommend using in-line styles on your component, as the day slot content is passed down multiple levels within the component. Another way to style the component is to target the `part="daySlot"` attribute that is added to the container around the slot content.
+
+**Disclaimer**: At this time, we are only supporting the use of text and `auro-icon` as the content being used in the slot.
+
+In this example, we use text content within the day slot to display prices for each day.
+
```
Comment (blackfalcon) [inline docs/partials/api.md:363]: Please see the requested updates to the `demo/api.md` document. Sorry, should have put them here, but ¯\_(ツ)_/¯ oops.

File: docs/partials/installDeps.md [hunk 7]
```diff
@@ -3,6 +3,6 @@
 The following Auro peer dependencies must be included.
 
 ```shell
-$ npm i alaskaairux/design-tokens
+$ npm i aurodesignsystem/design-tokens
```
Comment (blackfalcon) [inline docs/partials/installDeps.md:6]:
```suggestion
$ npm i @aurodesignsystem/design-tokens
```

File: scripts/generateDocs.js [hunk 8]
```diff
@@ -47,8 +47,8 @@ function formatTemplateFileContents(content, destination) {
    * Replace placeholder strings
    */
   result = result.replace(/\[npm]/g, nameExtractionData.npm);
-  result = result.replace(/\[name]/g, nameExtractionData.name);
-  result = result.replace(/\[Name]/g, nameExtractionData.nameCap);
+  result = result.replace(/\[name](?!\()/g, nameExtractionData.name);
+  result = result.replace(/\[Name](?!\()/g, nameExtractionData.nameCap);
```
Comment (blackfalcon) [inline scripts/generateDocs.js:51]: What is the reason for this change? How does this change the output of the generated docs? Do we need to make this update to the Auro Generator? Or did this come from the Generator?

File: src/auro-calendar-month.js [hunk 9]
```diff
@@ -10,9 +10,20 @@ import './auro-calendar-cell';
...
+
+    this.slotMap = new Map();
+  }
+
+  // This function is to define props used within the scope of this component
+  // Be sure to review  https://lit-element.polymer-project.org/guide/properties#reflected-attributes
+  // to understand how to use reflected attributes with your property settings.
+  static get properties() {
+    return {
+      slotMap:       { type: Map }
```
Comment (blackfalcon) [inline src/auro-calendar-month.js:24]:
```suggestion
      slotMap: { type: Map }
```

File: src/auro-calendar-month.js [hunk 10]
```diff
@@ -58,6 +70,61 @@ export class AuroCalendarMonth extends RangeDatepickerCalendar {
...
+  }
+
+  insertSlotContentByDay() {
+    const renderedDays = [...this.shadowRoot.querySelectorAll('auro-calendar-cell')];
+
+    renderedDays.forEach((dateCell) => {
+      const day = new Date(dateCell.day.date * 1000).getDate();
+      const dayName = this.getFormattedDate(dateCell);
+
+      if (this.slotContentByDay && this.slotContentByDay[day] && this.slotContentByDay[day][0].content.getAttribute('date') === dayName) {
```
Comment (blackfalcon) [inline src/auro-calendar-month.js:101]: This `if` statement is pretty procedural. Could this be optimized to be a function instead?

File: src/auro-calendar.js [hunk 11]
```diff
@@ -1,6 +1,8 @@
 import { html } from 'lit';
 import styleCss from "./style-auro-calendar-css";
 
+import 'lodash';
```
Comment (blackfalcon) [inline src/auro-calendar.js:4]: WIth lodash being added to the repo as a dependency can we document why and how this is being used?

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: scripts/generateDocs.js (hunk 8 above)
Comment (jordanjones243) [reply scripts/generateDocs.js:51]: This update came from the Generator

File: demo/api.md (hunk 1 above)
Comment (jordanjones243) [reply demo/api.md:747]: Unfortunately, an example using css parts cannot be added at this time. Due to the way the slot content is getting passed down to the auro-calendar-cell level on render, adding a css rule within the html for the specific part attribute does not target the attribute correctly, nor apply the style.

That being said, I completely agree with your requested change, I am just not sure how to correctly implement the example at this time.

**Review by jason-capsule42 — APPROVED** [review]
